blockchain/types.go

```diff
@@ -9,9 +9,29 @@ import (
 type Event struct {
 	Data        interface{}
 	BlockNumber uint64
+	TxIndex     uint64
+	Index       uint64
 	TS          uint64
 }
 
+func (m *Event) PrecedesOrEquals(other *Event) bool {
+	var (
+		byBlockNumber = func(x, y *Event) bool { return x.BlockNumber < y.BlockNumber }
```

Comment:
Why don't you create a slice of functions so you don't have to repeat their names in the range?

```go
	a := []func(int){func (a int) { fmt.Println(a + 1) }, func (a int) { fmt.Println(a + 2) }}
	for _, f := range a {
		f(1)
	}
```

Reply:
good point